src/psyclone/tests/parse/module_info_test.py

```diff
@@ -347,11 +366,117 @@ def test_module_info_get_symbol(tmpdir, monkeypatch):
 end function myfunc1
 end module my_mod''')
 
-    minfo = ModuleInfo("my_mod", FileInfo(filepath))
-    assert isinstance(minfo.get_symbol("myfunc1"), RoutineSymbol)
+    module_info: ModuleInfo = ModuleInfo("my_mod", FileInfo(filepath))
+    assert isinstance(module_info.get_symbol("myfunc1"), RoutineSymbol)
     # A Symbol that doesn't exist.
-    assert minfo.get_symbol("amos") is None
-    # When no Container has been created. Monkeypatch get_psyir() to simplify
+    assert module_info.get_symbol("amos") is None
+    # When no Container has been created. Monkeypatch
+    # get_psyir() to simplify
     # this.
-    monkeypatch.setattr(minfo, "get_psyir", lambda: None)
-    assert minfo.get_symbol("amos") is None
+
+    def raise_error():
+        from psyclone.parse.file_info import FileInfoFParserError
+        raise FileInfoFParserError("Dummy error")
+
+    monkeypatch.setattr(
+        module_info,
+        "get_psyir",
+        raise_error)
+    monkeypatch.setattr(
+        module_info,
+        "get_psyir",
+        raise_error)
+    assert module_info.get_symbol("amos") is None
+
+
+def test_module_info_coverage_a(tmpdir, monkeypatch):
+    """
+    Coverage test
+    """
+
+    filename = os.path.join(tmpdir, "testfile_module_info_coverage.f90")
+
+    #
+    # Get fparser
+    #
+    with open(filename, "w", encoding='utf-8') as fout:
+        fout.write(SOURCE_DUMMY)
+
+    # We create a dummy
+    node = Container("Dummy")
+    module_info: ModuleInfo = ModuleInfo(
+            "my_mod",
+            FileInfo(filename),
+            node
+        )
+
+    # view() creates also a list of all modules based on fparser!!!
+    module_info.view_tree()
+    assert module_info is not None
+
+
+def test_module_info_coverage_source_node_found(tmpdir, monkeypatch):
+    """
+    Coverage test: File not found
+    """
+
+    node = Container("Dummy")
+    module_info: ModuleInfo = ModuleInfo(
+            "my_mod",
+            FileInfo("/tmp/source_not_found.f90"),
+            node
+        )
+
+    with pytest.raises(ModuleInfoError) as einfo:
+        module_info.get_source_code()
+
+    assert "Could not find file" in str(einfo.value)
+
+
+def test_module_info_coverage_fparser_tree(tmpdir, monkeypatch):
```

Comment:
And again...

Reply:
OK. I also split this into two individual tests, see the one below.

Reply:
Good stuff, thanks.